File: leetcode/palindrome_number/go/solve.go
package main

import (
	"fmt" 
	"strconv"
)


func isPalindrome(x int) bool {
    strX := strconv.Itoa(x)
	rev := []rune{}
	for i := len(strX) - 1; i >= 0; i-- {
		rev = append(rev, rune(strX[i]))
	} 
	newStr := string(rev)
	return strX == newStr
}

func main() {
	fmt.Println(isPalindrome(121))
	fmt.Println(isPalindrome(-121))
	fmt.Println(isPalindrome(10))
	fmt.Println(isPalindrome(101))
}

File: leetcode/palindrome_number/javascript/solve.js
/**
 * @param {number} x
 * @return {boolean}
 */
const isPalindrome = function(x) {
    const strX = x.toString()
    let rev = "";
    for (let i = strX.length - 1; i >= 0; i--) rev += strX[i];
    return strX === rev;
};

console.log(isPalindrome(121));
console.log(isPalindrome(-121));
console.log(isPalindrome(10));
console.log(isPalindrome(101));